pyblish/plugin.py

```diff
@@ -462,10 +503,38 @@ def process(plugin, context, instance=None, action=None):
 
     """
 
-    if issubclass(plugin, (ContextPlugin, InstancePlugin)):
-        result = __explicit_process(plugin, context, instance, action)
-    else:
-        result = __implicit_process(plugin, context, instance, action)
+    if not context.get('profiling'):
+        context.data['profiling'] = {}
+
+    profiler = ProfilingTimer()
+    with profiler:
+        if issubclass(plugin, (ContextPlugin, InstancePlugin)):
+            result = __explicit_process(plugin, context, instance, action)
+        else:
+            result = __implicit_process(plugin, context, instance, action)
+
+    profiling_data = {
+        "runtime": profiler.runtime,
+    }
+
+    # collected using psutil module, None if it is not installed
+    try:
+        profiling_data['initial_memory_sage'] = profiler.initial_mem
```

Comment:
Typo on `_sage`. The naming convention should be `mixedCase` for dictionary keys as well (JSON convention).